test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ExtensionWithDispatchQueue.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                get { return _bindingContract; }
+            }
+
+            public Type TriggerValueType
+            {
+                get { return typeof(JObject); }
+            }
+
+            public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+            {
+                JObject triggerValue = value as JObject;
+                IValueProvider valueBinder = new ExtensionValueBinder(triggerValue);
+                return Task.FromResult<ITriggerData>(new TriggerData(valueBinder, GetBindingData(triggerValue)));
+            }
+
+            public Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
+            {
+                return Task.FromResult<IListener>(new Listener(context));
+            }
+
+            public ParameterDescriptor ToParameterDescriptor()
+            {
+                return new ExtensionTriggerParameterDescriptor
+                {
+                    Name = _parameter.Name,
+                    DisplayHints = new ParameterDisplayHints
+                    {
+                        Prompt = "DispatchQueue",
+                        Description = "DispatchQueue trigger fired",
+                        DefaultValue = "null"
+                    }
+                };
+            }
+
+            private IReadOnlyDictionary<string, object> GetBindingData(JObject value)
+            {
+                Dictionary<string, object> bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                bindingData.Add("DispatchQueueTrigger", value);
+                return bindingData;
+            }
+
+            private IReadOnlyDictionary<string, Type> CreateBindingDataContract()
+            {
+                Dictionary<string, Type> contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+                contract.Add("DispatchQueueTrigger", typeof(JObject));
+                return contract;
+            }
+
+            private class ExtensionTriggerParameterDescriptor : TriggerParameterDescriptor
+            {
+                public override string GetTriggerReason(IDictionary<string, string> arguments)
+                {
+                    return string.Format("DispatchQueue trigger fired at {0}", DateTime.Now.ToString("o"));
+                }
+            }
+
+            private class ExtensionValueBinder : IValueProvider
+            {
+                private readonly object _value;
+
+                public ExtensionValueBinder(JObject value)
+                {
+                    _value = value;
+                }
+
+                public Type Type => typeof(JObject);
+
+                public Task<object> GetValueAsync()
+                {
+                    return Task.FromResult(_value);
+                }
+
+                public string ToInvokeString()
+                {
+                    return _value.ToString();
+                }
+            }
```

Comment:
We have this helper in several places. See ObjectValueProvider

Reply:
that's an internal class, I can't use it easily in end to end test